tests/installation/bootloader.pm

```diff
@@ -85,51 +85,40 @@ sub run() {
 
     # set HTTP-source to not use factory-snapshot
     if (get_var("NETBOOT")) {
+        my $m_protocol = get_var('INSTALL_SOURCE', 'http');
+        my ($m_server, $m_share, $m_directory);
+
+        # Parse SUSEMIRROR into variables
+        if (get_var("SUSEMIRROR", '') =~ m{^([a-zA-Z0-9.-]*)(/.*)$}) {
+            ($m_server, $m_directory) = ($1, $2);
+            if ($m_protocol eq "smb") {
+                ($m_share, $m_directory) = $m_directory =~ /\/(.+?)(\/.*)/;
+            }
+        }
+
+        # select installation source (http, ftp, nfs, smb)
         send_key "f4";
         assert_screen "inst-instsourcemenu";
-        # select a net installation source (http, ftp, nfs, smb) by using send_key_until_needlematch
-        send_key_until_needlematch 'inst-instsourcemenu-' . get_var('INSTALL_SOURCE', 'http'), 'down';
+        send_key_until_needlematch 'inst-instsourcemenu-' . $m_protocol, 'down';
```

Comment:
I would do this as `send_key_until_needlematch "inst-instsourcemenu-$m_protocol", 'down'`

ie. I'd use double-quotes and perl variable expansion.. nothing wrong with this if it works though